### PR TITLE
MCO-392: Start using rhel-coreos image rather than machine-os-content.

### PR DESCRIPTION
**Pull request by ybettan**
The machine-os-content is deprecated and we'd like to stop shipping it entirely in the release payload. Point the DTK imagestream to the rhel-coreos image instead.

---

Closes: https://github.com/openshift/driver-toolkit/issues/101
/cc @jlebon

### CODE REVIEW

**Comment by jlebon** [comment]
Thanks for opening this!

> /hold
> Should be merged together with [openshift/release#48526](https://github.com/openshift/release/pull/48526)

It's not necessary they go together. https://github.com/openshift/release/pull/48526 should be able to go in alone. Once we confirm everything still works fine, and we're confident there's nothing else relying on it, then we can coordinate on flipping over again DTK, RHCOS and ART together.

**Comment by ybettan** [comment]
> It's not necessary they go together. https://github.com/openshift/release/pull/48526 should be able to go in alone. Once we confirm everything still works fine, and we're confident there's nothing else relying on it, then we can coordinate on flipping over again DTK, RHCOS and ART together.

Ack. I am not sure what https://github.com/openshift/release/pull/48526 really does but in DTK we are referencing to the `machine-os-content` labels for the `driver-toolkit` imagestream in the cluster (which is used by KMM to build kmods in cluster).

Anyway, I will keep it on hold until we have the ART+RHCOS+DTK parts together.
Thanks for the update.

**Comment by jlebon** [comment]
OK, we're looking good now. openshift/kubernetes references have been removed.
I've re-staged the openshift/os change in https://github.com/openshift/os/pull/1457.
Let's target early next week to get these in? Will reach out more directly for coordinating.

File: manifests/01-openshift-imagestream.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ spec:
     from:
       kind: DockerImage
       name: example.com/image-reference-placeholder:driver-toolkit
-  - name: 0.0.1-snapshot-machine-os 
+  - name: 0.0.1-snapshot-rhel-coreos
```
Comment (jlebon) [inline manifests/01-openshift-imagestream.yaml:18]: Are you sure this change is required? IIUC, `machine-os` is a reference to whatever image has the `machine-os` component name, i.e. these kinds of labels:

```
"io.openshift.build.version-display-names": "machine-os=CentOS Stream CoreOS",
"io.openshift.build.versions": "machine-os=416.94.202403121338-0",
```

And that'll correctly refer to the new `rhel-coreos` image once https://github.com/openshift/os/pull/1457 is in.

([The previous stab at this](https://github.com/openshift/driver-toolkit/pull/135) didn't change this and IIRC I don't _think_ we hit issues related to that in the brief period where we had flipped all the switches.)

Comment (ybettan) [reply manifests/01-openshift-imagestream.yaml]: I think you are right. I have reverted this part.

**Comment by ybettan** [comment]
/override ci/prow/e2e-aws-driver-toolkit-presubmit ci/prow/e2e-upgrade

**Pull request merged**